FirebaseAI/Tests/Unit/Types/ToolTests.swift

```diff
@@ -0,0 +1,51 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import XCTest
+
+@testable import FirebaseAI
+
+@available(iOS 15.0, macOS 12.0, macCatalyst 15.0, tvOS 15.0, watchOS 8.0, *)
+final class ToolTests: XCTestCase {
+  let encoder = JSONEncoder()
+
+  override func setUp() {
+    super.setUp()
+    encoder.outputFormatting = .sortedKeys
+  }
+
+  func testEncodeTool_googleSearch() throws {
+    let tool = Tool.googleSearch()
+    let jsonData = try encoder.encode(tool)
+    let jsonString = try XCTUnwrap(String(data: jsonData, encoding: .utf8))
+    XCTAssertEqual(jsonString, "{\"googleSearch\":{}}")
+  }
+
+  func testEncodeTool_functionDeclarations() throws {
+    let functionDecl = FunctionDeclaration(
+      name: "test_function",
+      description: "A test function.",
+      parameters: ["param1": .string()]
+    )
+    let tool = Tool.functionDeclarations([functionDecl])
+
+    encoder.outputFormatting.insert(.withoutEscapingSlashes)
+    let jsonData = try encoder.encode(tool)
+    let jsonString = try XCTUnwrap(String(data: jsonData, encoding: .utf8))
+
+    XCTAssertEqual(jsonString, """
+    {"functionDeclarations":[{"description":"A test function.","name":"test_function","parameters":{"nullable":false,"properties":{"param1":{"nullable":false,"type":"STRING"}},"required":["param1"],"type":"OBJECT"}}]}
+    """)
```

Comment:
I think this may be more readable if we use `encoder.outputFormatting = [.prettyPrinted, .sortedKeys, .withoutEscapingSlashes]`. This file shows the pretty print formatting style that Apple uses:
https://github.com/firebase/firebase-ios-sdk/blob/2c45a6c1596eb8ba60082301e55cc87c71f43e22/FirebaseAI/Tests/Unit/Types/SchemaTests.swift#L25-L41